compiler/src/dotty/tools/dotc/ast/DesugarEnums.scala

```diff
@@ -117,9 +117,11 @@ object DesugarEnums {
     val rawEnumClassRef = rawRef(enumClass.typeRef)
     extension (tpe: NamedType) def ofRawEnum = AppliedTypeTree(ref(tpe), rawEnumClassRef)
 
-    val lazyFlagOpt = if enumCompanion.owner.isStatic then EmptyFlags else Lazy
-    val privateValuesDef = ValDef(nme.DOLLAR_VALUES, TypeTree(), ArrayLiteral(enumValues, rawEnumClassRef))
-      .withFlags(Private | Synthetic | lazyFlagOpt)
+    val privateValuesDef =
+      val uncheckedValues =
+        Annotated(ArrayLiteral(enumValues, rawEnumClassRef), New(ref(defn.UncheckedAnnot.typeRef)))
+      ValDef(nme.DOLLAR_VALUES, TypeTree(), uncheckedValues)
+        .withFlags(Private | Synthetic)
```

Comment:
The `lazyFlagOpt` disappeared. Was this a mistake during conflict resolution?

Reply:
no, it was removed because it was only required when the enum values were explicitly selected from the companion, now they are selected from `this` which doesn't force the companion to be initialised.

Reply:
Ah, that makes sense!